src/app/components/turnos/dar-turnos/calendario-dia.class.ts

```diff
@@ -16,19 +16,28 @@ export class CalendarioDia {
     public delDiaDisponibles = 0;
     public profesionalDisponibles = 0;
 
-    constructor(public fecha: Date, public agenda: any, solicitudPrestacion: any) {
+    constructor(public fecha: Date, public agenda: any, solicitudPrestacion: any, filtroPrestacion?: any, filtroProfesional?: any) {
         this.hoy = new Date();
         this.turnosDisponibles = 0;
         if (!agenda) {
             this.estado = 'vacio';
         } else {
-            let disponible: boolean = (this.agenda.turnosDisponibles > 0);
+            let bloquesPrestacion = agenda.bloques; // para filtrado de agendas/bloques segun prestacion y/o profesional
+            if (filtroProfesional && !agenda.profesionales.find(prof => prof.id === filtroProfesional.id)) {
+                // si se está filtrando por un profesional que no está en la agenda ...
+                bloquesPrestacion = [];
+            }
+            if (filtroPrestacion) {
+                bloquesPrestacion = bloquesPrestacion.filter(b => b.tipoPrestaciones.find(tipo => tipo.id === filtroPrestacion.id));
+            }
+            let hayTurnosDisponibles: boolean = (bloquesPrestacion.find(b => b.turnos.find(t => t.estado === 'disponible')));
```

Comment:
```suggestion
            let hayTurnosDisponibles = bloquesPrestacion.some(b => b.turnos.some(t => t.estado === 'disponible'));
```
Le saco el tipo de dato porque se intuye desde el some